### PR TITLE
Enabling release version selection when deploying a solution

### PR DESCRIPTION
**Pull request by ppathan**
# Type of change? 

- [ ] Bug fix
- [x] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 
Currently for a released version of the pcs cli it is tied to a particular release version of the docker image. This change allows the user to specify a release version. The default release version is going to be "testing" which always points to latest.

**Checklist:**

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by dluc** [comment]
I understand why we would make "testing" the default option, at least when using the CLI from Github, however that will easily generate another sort of confusion when getting the CLI from npm repo.  I would prefer for the user to be explicit here, so we don't have user deploying a development version instead of a official version

imho the user should specify the version number via a parameter, and if they don't then the CLI should prompt. Similarly, the same for the solution to deploy, because soon we will have multiple solutions, so it won't make sense to default to Remote Monitoring.

on a bigger topic, I would like the CLI to support headless mode, allowing a user to pass all the parameters at launch, without prompt. When parameters are not passed, the CLI tool should prompt for these.

File: src/index.ts [hunk 1]
```diff
@@ -62,14 +62,15 @@ const MAX_RETRYCOUNT = 36;
...
-const program = new Command(packageJson.name)
+const program = new Command('pcs')
     .version(packageJson.version, '-v, --version')
     .option('-t, --type <type>', 'Solution Type: remotemonitoring', /^(remotemonitoring|test)$/i, 'remotemonitoring')
     .option('-s, --sku <sku>', 'SKU Type: basic, standard, or test', /^(basic|standard|test)$/i, 'basic')
     .option('-e, --environment <environment>',
             'Azure environments: AzureCloud or AzureChinaCloud',
             /^(AzureCloud|AzureChinaCloud)$/i, 'AzureCloud')
     .option('-r, --runtime <runtime>', 'Microservices runtime: dotnet or java', /^(dotnet|java)$/i, 'dotnet')
+    .option('--release <release>', 'Microservices release version: currently published or testing', /^[\w\.\-\_]{1,64}$/i, 'testing')
```
Comment (timlaverty) [inline src/index.ts]: I don't think we should allow the user to enter an arbitrary string - I think we need to offer a selection of known releases that and ask the user to select amongst them.  I also think we should consider whether to lookup the appropriate tags underneath that release per container or whether the tags will always be consistent.

Comment (ppathan) [reply src/index.ts]: If we are just talking about versions that we released sure it makes sense to show a list of releases, but if there are releases that are outside of what we pushed to docker we can't possibly know their values. Anyways lets chat offline.

**Comment by dluc** [comment]
Something like this could work:

What IoT solution would you like to deploy?
[ ] Remote Monitoring, latest release
[ ] Remote Monitoring, development build
[ ] xyz, latest release
[ ] xyz, development build

Which SKU?
[ ] Standard   <-- preselected
[ ] Basic

Which language?
[ ] .NET
[ ] Java


with the option to skip the questions using parameters, e.g.
```
pcs --solution remotemonitoring --release 1.0.0 --sku standard --language dotnet 
```

**Comment by ppathan** [comment]
Let us talk first and think through the options more carefully. I'd rather avoid design a new feature here if possible.

File: remotemonitoring/scripts/all-in-one.yaml [hunk 2]
```diff
@@ -194,7 +194,7 @@ spec:
     spec:
       containers:
       - name: device-simulation-pod
-        image: azureiotpcs/device-simulation-dotnet:testing
+        image: azureiotpcs/device-simulation-dotnet:{release}
```
Comment (dluc) [inline remotemonitoring/scripts/all-in-one.yaml]: Docker image tags are going to diverge, e.g. we'll have 
* 1.0.0-preview.2
* 1.0.3-preview.2
as soon as someone will send a PR.
Can we use the release number to identify a docker-compose and k8s configuration instead?

Comment (dluc) [reply remotemonitoring/scripts/all-in-one.yaml]: e.g. something like
* docker-compose.testing.dotnet.yaml
* docker-compose.1.0.0-preview.1.dotnet.yaml
* docker-compose.1.0.0-preview.2.dotnet.yaml
etc.

File: remotemonitoring/single-vm/setup.sh [hunk 3]
```diff
@@ -73,9 +74,138 @@ cd ${APP_PATH}
...
-DOCKERCOMPOSE_SOURCE="${REPOSITORY}/docker-compose.${APP_RUNTIME}.yml"
-wget $DOCKERCOMPOSE_SOURCE -O ${DOCKERCOMPOSE}
+touch ${DOCKERCOMPOSE} && chmod 550 ${DOCKERCOMPOSE}
+echo "version: '2'"                                                                         >> ${DOCKERCOMPOSE}
+echo ""                                                                                     >> ${DOCKERCOMPOSE}
+echo "services:"                                                                            >> ${DOCKERCOMPOSE}
+echo "  reverseproxy:"                                                                      >> ${DOCKERCOMPOSE}
+echo "    image: azureiotpcs/remote-monitoring-nginx:${APP_RELEASE_VERSION}"                >> ${DOCKERCOMPOSE}
+echo "    ports:"                                                                           >> ${DOCKERCOMPOSE}
+echo "      - \"80:80\""                                                                    >> ${DOCKERCOMPOSE}
```
Comment (dluc) [inline remotemonitoring/single-vm/setup.sh]: btw, we should remove port 80 before public preview

Comment (ppathan) [reply remotemonitoring/single-vm/setup.sh]: I can remove it now if it is not needed, i wasn't sure.

File: src/index.ts [hunk 4]
```diff
@@ -160,6 +161,20 @@ function main() {
...
             } else {
                 const questions: IQuestions = new Questions(program.environment);
+                // If user didn't specify any release version then prompt from the version
+                // that have been released so far. Also note that this list will need to be
+                // updated with each release. More long term solution would be to look at github
+                // tags/releases to populate the list of options shown to the user per solution
+                // type.
+                if (!program.release) {
+                    questions.addQuestion({
+                        choices: [{value: '1.0.0-preview'}, {name: 'latest', value: 'testing'}],
```
Comment (dluc) [inline src/index.ts]: can we use a different name for the development build? "latest" sounds like a stable release, I would suggest "last development build" or similar

File: remotemonitoring/scripts/all-in-one.yaml (hunk 2 above)
Comment (ppathan) [reply remotemonitoring/scripts/all-in-one.yaml]: I would suggest for now based on our discussion earlier that we rollover version numbers for all the images as long as we are in preview. We could use the release number on 'RemoteMonitoring' solution to figure out what individual docker image versions for the microservices are. Not sure if that is what you meant by using release number.

File: src/index.ts (hunk 4 above)
Comment (ppathan) [reply src/index.ts]: i could just call it 'development'

**Comment by ppathan** [comment]
Based on discussion offline, closing this PR. Post public preview we will work on more complete fix that is long term.

**Pull request closed without merging**